pkg/runtime/grpc/oci.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcruntime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func (r *Runtime) GetOCIGadgetInfo(gadgetCtx runtime.GadgetContext) (*api.GadgetInfo, error) {
+	timeout := time.Second * time.Duration(r.globalParams.Get(ParamConnectionTimeout).AsUint16())
+	ctx, cancelDial := context.WithTimeout(gadgetCtx.Context(), timeout)
+	defer cancelDial()
+
+	// use default params for now
+	params := r.ParamDescs().ToParams()
+	conn, err := r.getConnToRandomTarget(ctx, params)
+	if err != nil {
+		return nil, fmt.Errorf("dialing random target: %w", err)
+	}
+	defer conn.Close()
+	client := api.NewOCIGadgetManagerClient(conn)
+
+	paramMap := make(map[string]string)
+
+	// OCI exists both on client and server
```

Comment:
Mmmm I'm not sure I understand this comment. What does it do on the client side?